src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -644,6 +627,30 @@ internal static bool Dispatch()
         }
     }
 
+    // Simple random number generator. We don't need great randomness, we just need a little and for it to be fast.
+    internal struct FastRandom // xorshift prng
+    {
+        private uint _w, _x, _y, _z;
+
+        public FastRandom(int seed)
+        {
+            _x = (uint)seed;
+            _w = 88675123;
+            _y = 362436069;
+            _z = 521288629;
+        }
+
+        public int Next(int maxValue)
+        {
+            uint t = _x ^ (_x << 11);
+            _x = _y; _y = _z; _z = _w;
+            _w = _w ^ (_w >> 19) ^ (t ^ (t >> 8));
+
+            int r = (int)_w & int.MaxValue;
+            return (int)(r * (1.0 / ((double)int.MaxValue + 1)) * maxValue);
```

Comment:
I still wonder if light coordination between the threads using interlocked might be better than random to avoid collisions

As in rather than
`int i = tl.random.Next(c);`
do
`int i = NextCounter.Next(c);`

from https://gist.github.com/benaadams/9a66811bc54e7126a9cd8c181bbe6f53

Maybe something to look at in future.